includes/class-wcpdf-admin.php

```diff
@@ -786,18 +720,17 @@ public function regenerate_document() {
 			) );
 		}
 
-		$form_data = json_decode(stripslashes( $_POST['json_data'] ), true);
-
-		$order_id = absint( $form_data['order_id'] );
-		$document = sanitize_text_field( $form_data['document'] );
+		$order_id = absint( $_POST['order_id'] );
+		$document_type = sanitize_text_field( $_POST['document_type'] );
 
 		try {
-			$document = wcpdf_get_document( $document, wc_get_order( $order_id ) );
+			$document = wcpdf_get_document( $document_type, wc_get_order( $order_id ) );
 			if ( !empty($document) && $document->exists() ) {
 				// save number and date
-				$this->save_invoice_number_date($order_id, null);
+				$this->save_invoice_number_date( $order_id, null );
```

Comment:
shouldn't this be removed?

Reply:
Number is not being saved, are you sure i don't need this?

Reply:
If the number is not being saved, then your `set_data` below this is not working properly. But at this point I can't tell because you have resolved all conversations without pushing the commits that resolved it 😄

Reply:
I know, sorry, let me try debug quickly to see if i can find the "bug".